File: CSharp/PaymentsService/Domain/Payments/ProcessPaymentBaseRequest.cs
namespace PaymentsService.Domain.Payments;

public abstract class ProcessPaymentBaseRequest
{
    public ProcessPaymentBaseRequest(string paymentMethod)
    {
        PaymentMethod = paymentMethod;
    }

    public string PaymentMethod { get; set; }
    public string Currency { get; set; }
    public decimal Amount { get; set; }
}

File: Python/PaymentsService/Domain/Payments/ProcessPaymentBaseRequest.py
from abc import ABC, abstractmethod

class ProcessPaymentBaseRequest(ABC):
    def __init__(self, payment_method, currency = None, amount = None):
        self.paymentMethod = payment_method
        self.currency = currency
        self.amount = amount
    
    @abstractmethod
    def to_dict(self) -> dict:
        pass